peer/peer.go

```diff
@@ -370,24 +373,28 @@ func exchangeListenAddrs(c net.Conn, d time.Duration) (string, error) {
 		receivedAddr := false
 		sentAddr := false
 		var addr string
+		var ok bool
 
 		for !receivedAddr || !sentAddr {
-			message, err := msg.Read(c)
+			payload, err := msg.Read(c)
```

Comment:
Minor thing but I think calling this `message` still makes sense. Technically we're extracting the "payload" from the "message" in the message package but from the context of peer this is a message.